### DISCUSSION TITLE
AOSP 12.1 v411

### DISCUSSION THREAD

**Discussion by phhusson** [phhusson/treble_experimentations#2236] [General]
- Fixes boot on some devices (notably Mediatek but probably others as well)
- Fixes gapps variant' launcher
- Fixes floss' variant webview
- Fix setup wizard on gapps variant
- Fix fingerprint on some oplus devices (Thanks @pjgowtham)

<hr /><em>This discussion was created from the release <a href='https://github.com/phhusson/treble_experimentations/releases/tag/v411'>AOSP 12.1 v411</a>.</em>

**Comment by JonnyTech** [comment]
Will the webview fix be backported to v3?

**Reply by phhusson** [reply]
It technically already is, but I don't plan on rebuilding a v3xx anytime soon

**Reply by JonnyTech** [reply]
Understood - can I patch my v3 install without rebuilding? Is there a commit that I can look at (cannot find it myself)?

**Reply by phhusson** [reply]
@JonnyTech actually yes you can, take v411's system/product/overlay/treble-overlay-highpriomisc/treble-overlay-highpriomisc.apk and put it in your /system/product/overlay, and then install bromite webview apk "normally" and it should work

**Reply by spiceke** [reply]
I think the download links are broken..does not pass 50%

**Reply by JonnyTech** [reply]
> @JonnyTech actually yes you can, take v411's system/product/overlay/treble-overlay-highpriomisc/treble-overlay-highpriomisc.apk and put it in your /system/product/overlay, and then install bromite webview apk "normally" and it should work

Thank you - by normally I assume that the existing `/system/product/app/webview/webview.apk` also needs replacing - I shall test in the next day or two and report back.

**Reply by phhusson** [reply]
No it doesn't. Just add the overlay apk to /system, and install bromite  webview apk just like any apk, and you'll be able to select bromite webview in developer settings

**Comment by carlosehidalgo** [comment]
Excellent continuous update, really great work, in Samsung Z Flip 3 it does not start, the equipment restarts without the possibility of starting the ROM, it would be fantastic to have this ROM for the new trend of mobile equipment. Thanks

**Reply by phhusson** [reply]
Feel free to offer me one. If I get one, I definitely can make it boot.
(Usable is another story though)

**Reply by carlosehidalgo** [reply]
You have my device for it, I have DSU installed to do tests, let me know and I will gladly help you with whatever you need.

**Comment by iviliprins** [comment]
There is no animation in gestures  for lift to check phone and quickly open 📸 navigation its ok, :(

**Comment by steven121992** [comment]
I flashed the secure img but my device is still not certified. I registered on google and rebooted but still no luck. I even flashed the universal safetynet fix. Some of my banking apps aren't even launching.

**Reply by phhusson** [reply]
Which GSI variant? Which device?

To fix this, register your device on https://www.google.com/android/uncertified/

To get the ID, you can either:
- adb root, then adb shell, press enter, and at the new prompt, inside adb shell, do `sqlite3 /data/data/com.google.android.gsf/databases/gservices.db    'select * from main where name = "android_id";'`

**Reply by steven121992** [reply]
Samsung m20. Vndklite gapps secure. I did register. But it still shows device uncertified.

**Comment by luiz-gadelha** [comment]
Hello, I have a K41s, in other versions I was able to enable the fingerprints but in this version I can't, is there a solution?

**Reply by phhusson** [reply]
In setup wizard, in settings or both?
What happens when you try to?

**Reply by luiz-gadelha** [reply]
happens to everyone. I try more settings and it closes the application.

**Reply by luiz-gadelha** [reply]
https://user-images.githubusercontent.com/97181241/159578483-aac272a3-e85c-4d94-a857-55280e74e684.mp4

**Reply by steupz** [reply]
This happened to me when I installed the GoGapps variant. But fingerprint works with the Non gapp.

**Comment by der-lange** [comment]
great progress, thank you!
But...... broken camera: Sony Xperia XZ2 Premium (Bionic driver problem?)

**Reply by der-lange** [reply]
This link may help
https://forum.xda-developers.com/t/xz2p-on-gsi-11-12-android.4421169/#post-86629807
but unfortunetly trying with Magisk-v24.3 (or alternatively Magisk-v21.4)
Error: "non supported Magisk-version"
Can you give any advice?

**Reply by der-lange** [reply]
Even the original SONY Xperia camera is working now fine! Have a look at the link above. You will find a "howto" there.  It´s worth while!

**Comment by Aleksandar1996** [comment]
Is OTA working on this version?

**Reply by luiz-gadelha** [reply]
why the animations, the shapes in the vanilla version are rounder than in the Gapps version?

**Reply by tnolte1** [reply]
Sadly, for me it's not working (Blackview A80Plus, Helio P22)

In updater I get:

System update available
Android Version: 12
GSI variant: treble_arm64_bgS-secure
Security patch: November 5, 2020

Update version: v411
Download size: 894 kB

Then when I try to install for some seconds:

Preparing storage for OTA...

Then I get a popup:

Error:
New update failed to be installed.
Check your logs for more information

The last lines from Logcat Extreme:

I Gralloc4: mapper 4.x is not supported
W Gralloc3: mapper 3.x is not supported
W Gralloc4: allocator 4.x is not supported
W Gralloc3: allocator 3.x is not supported
E ion: ioctl c0044901 failed with code -1: Invalid argument

**Comment by HeroRickyGAMES** [comment]
I tested it on my Moto G6 ALI, and I found a variety of bugs, one of these bugs is old, the camera doesn't take a picture (I believe it's a vendor bug) I also saw a bug in the boot that the bootanimation is not visible to me.

**Comment by nation888** [comment]
GSI install summary on Oukitel K9 A/B, 64bit Gapps, Secured
Specs MediaTek Helio P35 (MT6765), 64bit, 4GB RAM
Method: TWRP 3.3.1 system image clean install

Initial setup 

i. "Speech Services by Google keeps stopping" dialogue popped up
ii. Google restore asked repeated prompting error to close, initial Google restore repeats after credential acceptance. Cold boot repeats setup loop after trying repeatedly to by-pass Google setup. 
- Solution
1) Reverted to initial flash and "offline" setup up, skipped login, boot to home
2) Select Finish setup, connected to WIFI to complete account setup, and restore
3) New device Secure Setup (Method pattern), repeated secure confirmation error. Close setup and allow the restore process to continue
4) Magisk install no problems, Lposed -- riru no problems
5) Setup re-occurs to finish but stuck setup: solution after restoration disable Android Setup (System apps). Re-enabled, fully restored and eventually went away

Conclusion; Besides the google thing, everything else seems to work, and seems smooth to operate. BT, WIFI, GPS, all sensors work. Worth the test. Thank you

**Comment by HansDus** [comment]
Hello @phhusson
(Problem with v411 on Samsung Galaxy A505FN)

Now after Two Days running this Release, i have to say the Problem is sever!
The Phone regularly Crashes as you are in the middle of something makes for example writing a Text literally like playing Russian Roulette! SO if you want to write a E-Mail and put a lot of tort in it, when it Crashes its gone, because the E-Mail app will restart and what you did is gone.

So i am Unhappy now :-( understandably....
What can be done to stop this?
Is there a Stable Release as far as you Know?

Cant go back to v402 neither to many Bugs concerning Program that don't work anymore.
(remember our earlier exchange on that)
So v411 fixed a lot but now there is only that very annoying Bug that it crashes regularly!!

Pleas also don't forget i need reassurance what this Log file contains and if it can be used to harm my Privacy ore security of my Device in any way shape or form when i post it Publicly here!?

Id rather send you a E-Mail when you reassure me you make sure it will not be used in any malicious form!
in this case i would know only you get to Analyze it.

Pleas Help i realy Like Your Google-free Custum-Rom

Saludos Hans

**Comment by HansDus** [comment]
Hello @phhusson 

Since i don't hear from you i did following Temporally Solution!
Reason being:
The Crash Problem is a real Problem and makes working with my Device a gamble!
No good when you work a lot with it...as you can imagine!

Here is may solution until a stable an reliable AOSP 12.0 is available!

1.) I fall back to My System Backup (maid one with TWRP) it was Android 10 and AOSP v221
2.) I flashed my System with ASOP 11.0 v313 (last Android 11.0 Release of you)

Nice and Stable and a better Android 10 feels almost the same!
Everything is running with no Hickups :+1:  :1st_place_medal: 

I will keep Following to see when you got another release....
Great job and i still love you're Rom as i can imagine its not easy to create them...
As soon i have The possibility i send you some contributions.

I hope that my Information and feedback is helpful to you ore anyone that sees them.
It gives you the Prospective of a Power-User that dose not know it all an just want to use You re Custom-Rom for Privacy Purposes
and to stand a little against Big Tech and Big-Bro

Saludos Hans &
God Bless..

**Comment by RodrigoSMontes** [comment]
Hello @phhusson 

I liked these release. I'm using a Moto E6 Plus.

The only problem I'm facing is GPS not working. 

I've installed the system-squeak-arm64-ab-vndklite-gapps-secure variant.

Thanks for your hard working!

Best Regards

---------

Note: GPS is working after root. Thanks!

**Comment by BSTH1796** [comment]
I want to use Magisk V23 
How to remove the root that comes with the system

**Comment by luiz-gadelha** [comment]
I know, but the gapps module doesn't work in vanilla, Google play services
doesn't work and doesn't open

Em ter., 22 de mar. de 2022 23:03, steupz ***@***.***>
escreveu:
